### PR TITLE
Remove unwanted warning log for empty open-api definition

### PR DESCRIPTION
**Pull request by ayeshLK**
Fixes [#3613](https://github.com/ballerina-platform/ballerina-standard-library/issues/3613)

### CODE REVIEW

File: openapi-extension/src/main/java/io/ballerina/openapi/extension/doc/gen/AbstractOpenApiDocGenerator.java [hunk 1]
```diff
@@ -157,8 +155,10 @@ private void generateOpenApiDoc(OpenApiDocConfig config, SyntaxNodeAnalysisConte
                 config.getServiceNode(), listenerNodes, config.getSemanticModel(), targetFile, null);
         Optional<OpenAPI> openApiOpt = oasResult.getOpenAPI();
         if (!oasResult.getDiagnostics().isEmpty() || openApiOpt.isEmpty()) {
-            OpenApiDiagnosticCode errorCode = OpenApiDiagnosticCode.OPENAPI_107;
-            updateCompilerContext(context, location, errorCode);
+            if (embed) {
```
Comment (chamil321) [inline openapi-extension/src/main/java/io/ballerina/openapi/extension/doc/gen/AbstractOpenApiDocGenerator.java:158]: Merge condition in L157 and L158

File: openapi-extension/src/main/java/io/ballerina/openapi/extension/doc/gen/AbstractOpenApiDocGenerator.java [hunk 2]
```diff
@@ -93,11 +93,8 @@ public void generate(OpenApiDocConfig config, SyntaxNodeAnalysisContext context,
                     updateOpenApiContext(context, serviceId, openApiDefinition, embed);
                 } else {
                     // generate open-api doc and update the context if the `contract` configuration is not available
-                    generateOpenApiDoc(config, context, location);
+                    generateOpenApiDoc(config, context, location, embed);
                 }
-            } else {
-                // generate open-api doc and update the context
-                generateOpenApiDoc(config, context, location);
```
Comment (chamil321) [inline openapi-extension/src/main/java/io/ballerina/openapi/extension/doc/gen/AbstractOpenApiDocGenerator.java:100]: Is this case already covered?

**Comment by lnash94** [comment]
Shall we add a test for this fix, please let me know in case I missed it. :)

File: openapi-extension/src/main/java/io/ballerina/openapi/extension/doc/gen/AbstractOpenApiDocGenerator.java (hunk 2 above)
Comment (ayeshLK) [reply openapi-extension/src/main/java/io/ballerina/openapi/extension/doc/gen/AbstractOpenApiDocGenerator.java:100]: In this case we do not generate open-api doc for now (cos unless it is manually enabled usiing open-api annotation it should not generate). So, removed this logic.

File: openapi-extension/src/main/java/io/ballerina/openapi/extension/doc/gen/AbstractOpenApiDocGenerator.java (hunk 1 above)
Comment (ayeshLK) [reply openapi-extension/src/main/java/io/ballerina/openapi/extension/doc/gen/AbstractOpenApiDocGenerator.java]: Refactor the logic with [5b0b7e4a20209b7e3274b100d2777543d694827e](https://github.com/ballerina-platform/openapi-tools/pull/1108/commits/5b0b7e4a20209b7e3274b100d2777543d694827e)

**Review by lnash94 — APPROVED** [review]
LGTM

**Review by chamil321 — APPROVED** [review]

**Pull request merged**